packages/block-editor/src/components/warning/index.native.js

```diff
@@ -6,23 +6,24 @@ import { View, Text } from 'react-native';
 /**
  * WordPress dependencies
  */
-import { Icon, withTheme } from '@wordpress/components';
+import { Icon } from '@wordpress/components';
+import { withPreferredColorScheme } from '@wordpress/compose';
 import { normalizeIconObject } from '@wordpress/blocks';
 
 /**
  * Internal dependencies
  */
 import styles from './style.scss';
 
-function Warning( { title, message, icon, iconClass, theme, useStyle, ...viewProps } ) {
+function Warning( { title, message, icon, iconClass, theme, getStylesFromColorScheme, ...viewProps } ) {
 	icon = icon && normalizeIconObject( icon );
 	const internalIconClass = 'warning-icon' + '-' + theme;
```

Comment:
`theme` here should probably be renamed to `preferredColorScheme`?
This should fix the Warning component icon color issue.

Reply:
Indeed, forgot that one thanks 👍 , will scan the code for other occurrences as well!